nvflare/app_common/psi/dh_psi/dh_psi_workflow.py

```diff
@@ -163,7 +163,7 @@ def pairwise_setup(self, ordered_sites: List[SiteSize]):
 
         bop = BroadcastAndWait(self.fl_ctx, self.controller)
         results = bop.multicasts_and_wait(
-            task_name=self.task_name, task_inputs=task_inputs, abort_signal=self.abort_signal
+            task_name=self.task_name, task_inputs=task_inputs, fl_ctx=self.fl_ctx, abort_signal=self.abort_signal
```

Comment:
This isn't perfect. No object should keep self.fl_ctx. Caller should provide fl_ctx. You can keep "engine", and then create new fl_ctx with self.engine.new_context().

We can change this in another PR.

Reply:
Agreed, can change in another PR since this involves more refactoring